saleor/product/templatetags/sort_by.py

```diff
@@ -0,0 +1,11 @@
+from django import template
+
+register = template.Library()
+
+
+@register.inclusion_tag('category/_sort_by.html', takes_context=True)
```

Comment:
I think the idea of context was misunderstood here. The context that you get in first argument when using `takes_context=True` is [the template context as of when the tag was called](https://docs.djangoproject.com/en/1.11/howto/custom-template-tags/#inclusion-tags). What a inclusion tag should return is the context for the snippet it is rendering, not the whole context of the template. I don't see any need for the context in this tag. Here you should simply return a dictionary with attributes data.

Reply:
done, but we need to pass request all the way through get_sort_by_url